sentry-core/src/scope/real.rs

```diff
@@ -160,6 +160,17 @@ impl Scope {
     /// Sets the transaction.
     pub fn set_transaction(&mut self, transaction: Option<&str>) {
         self.transaction = transaction.map(Arc::from);
+        if let Some(name) = transaction {
```

Comment:
nit: if you're switching to `if let` i think it makes more sense to move line 162 into this block as well to keep the logic the same which make reading easier.

Reply:
The line above should explicitly set `None` as well though